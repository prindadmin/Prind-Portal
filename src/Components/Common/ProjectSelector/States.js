export const QUIESCENT = 'QUIESCENT'
export const LOADING = 'LOADING'
export const LOADING_ERROR = 'LOADING_ERROR'
export const UPDATE_ERROR = 'UPDATE_ERROR'
export const PROJECT_TYPE_SELECTOR_OPEN = 'PROJECT_TYPE_SELECTOR_OPEN'
