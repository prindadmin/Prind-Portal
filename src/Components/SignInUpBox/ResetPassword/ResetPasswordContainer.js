import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'

import * as reducer from '../../../Reducers/auth'

import Component from './ResetPassword'

const mapStatetoProps = state => {
  return {
    auth: state.auth,
  }
}

const mapDispatchToProps = dispatch => {
  return {
    setNewPassword: (values, resolve, reject) => {
      dispatch(reducer.changePassword(values, resolve, reject))
    }
  }
}

export default withRouter(connect(mapStatetoProps, mapDispatchToProps)(Component))
