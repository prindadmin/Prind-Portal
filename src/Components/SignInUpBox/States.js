export const QUIESCENT = 'QUIESCENT'
export const LOADING = 'LOADING'
export const PASSWORD_RESET_SUCCESS = 'PASSWORD_RESET_SUCCESS'
export const PASSWORD_RESET_FAILED = 'PASSWORD_RESET_FAILED'
