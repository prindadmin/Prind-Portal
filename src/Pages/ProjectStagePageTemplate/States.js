export const QUIESCENT = 'QUIESCENT'

export const CONTENT_FETCH_IN_PROGRESS = 'CONTENT_FETCH_IN_PROGRESS'
export const CONTENT_FETCH_SUCCESSFUL = 'CONTENT_FETCH_SUCCESSFUL'
export const CONTENT_FETCH_FAILED = 'CONTENT_FETCH_FAILED'
