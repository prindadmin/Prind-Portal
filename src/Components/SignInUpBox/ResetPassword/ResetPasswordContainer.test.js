import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import CanUseWebP from '../../../Functions/CheckIfWebpSupported'
jest.mock('../../../Functions/CheckIfWebpSupported')

import {
  HashRouter as Router
} from "react-router-dom";
import ReactGA from 'react-ga';

import * as Component from './ResetPasswordContainer'
import * as States from '../../../States'

// https://www.robinwieruch.de/react-connected-component-test

const mockStore = configureStore([]);

var component;

beforeEach(() => {
  CanUseWebP.mockReturnValue(true)
  ReactGA.initialize('dummy', { testMode: true });
  const store = mockStore({
    auth: {
      info: {},
      error: {},
      isSignedIn: States.AUTH_UNKNOWN,
      isConfirmed: States.AUTH_UNKNOWN,
      hasSignedUp: States.AUTH_UNKNOWN,
      hasSentCode: States.AUTH_UNKNOWN,
      hasChangedPassword: States.AUTH_UNKNOWN,
      passwordResetRequired: States.AUTH_UNKNOWN
    }
  });

  const mockToggleVisibleForm = jest.fn()
  const mockHistoryPush = jest.fn()
  const props = {
    toggleVisibleForm: mockToggleVisibleForm,
    history: {
      push: mockHistoryPush,
    },
    location: {
      search: "client_id=fbss4knsc8gmgct526ci8kp3a&user_name=892107a6-b223-4774-8dff-0b6fe5414ffe&confirmation_code=587828",
      pathname: "reset-password?",
    }
  }

  component = renderer.create(
    <Provider store={store}>
      <Router>
        <Component.default {...props} />
      </Router>
    </Provider>
  );
});


it('should render with given state', () => {
  expect(component.toJSON()).toMatchSnapshot();
});
