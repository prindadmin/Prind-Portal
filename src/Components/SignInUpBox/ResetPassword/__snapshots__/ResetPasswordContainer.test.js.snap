// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with given state 1`] = `
Array [
  <div
    className="logo-container"
  >
    <a
      href="https://buildingim.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="BuildingIM logo"
        src="/images/logos/buildingim-logo.webp"
      />
    </a>
  </div>,
  <div
    className="welcome-text-heading"
  >
    Welcome
  </div>,
  <div
    className="welcome-text-body"
  >
    Please enter a new password for your account.
  </div>,
  <form
    className="sign-in-form form"
    id="reset-password-form"
    onSubmit={[Function]}
  >
    <input
      className="default"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <input
      className="default"
      id="confirmPassword"
      name="confirmPassword"
      onChange={[Function]}
      placeholder="Repeat Password"
      type="password"
      value=""
    />
    <div
      className="spacer"
    />
    <input
      className="submit-button"
      id="submit"
      onClick={[Function]}
      type="submit"
      value="Change Password"
    />
  </form>,
]
`;
